[Live Debugger] Add instructions for Go (limited preview) (#32935)
DataDog/documentation: content/en/tracing/live_debugger/_index.md

The following constraints apply to Live Debugger usage and configuration:

> - **Language support:** Live Debugger is available for the same runtime languages as [Dynamic Instrumentation][14].

Reviewer: ```suggestion
- **Language support:** Live Debugger supports the same runtime languages as [Dynamic Instrumentation][14].
```
